test/api_integration/apis/general/csp.js

```diff
@@ -32,8 +32,16 @@ export default function ({ getService }) {
     it('csp header does not allow all inline scripts', async () => {
       const response = await supertest.get('/app/kibana');
 
-      expect(response.headers['content-security-policy']).to.contain('script-src');
-      expect(response.headers['content-security-policy']).not.to.contain('unsafe-inline');
+      const header = response.headers['content-security-policy'];
+      const parsed = new Map(header.split(';').map(rule => {
+        const parts = rule.trim().split(' ');
+        const key = parts.splice(0, 1)[0];
+        return [key, parts];
+      }));
+
+      const scriptSrc = parsed.get('script-src');
+      expect(scriptSrc).to.be.an(Array);
+      expect(scriptSrc).not.to.contain('unsafe-inline');
```

Comment:
__note:__ I'm a bit concerned about check like this. Our parsing logic in this test may not work as we expect at some point (even with packages like `content-security-policy-parser`) and we'll get false negatives here. Is there any reason we don't want to use `expect().to.eql` with a full list of "directives" here? Is it because of the `nonce-some-unguessable-value`? If so, can we filter `startsWith('nonce-')` out before asserting?

Reply:
Good point. I initially was trying to keep this test similar to how it was originally written, but you've made a convincing argument. Changes will be forth-coming.